GMS Crash with WearOS Device and Wifi with Password

- Device: Pixel 5a
- Build number: SQ3A.220705.003.A1.2022080300 (although this is not an new issue - I didn't track when this problem started, but presumably it's been happening since I paired the watch)
- I use Sandboxed Google Play Services, and the Google and WearOS apps are installed. I have a watch paired to my phone. Google Play Services has the Network permission.

On my Pixel 5a that is paired to a WearOS device, joining a wifi network with credentials causes a fatal GMS crash. Logcat here: [crash.txt](https://github.com/GrapheneOS/os-issue-tracker/files/9270252/crash.txt). Relevant excerpt:
```
08-05 09:14:51.779 17365 17365 E AndroidRuntime: java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.wifi.CONFIGURED_NETWORKS_CHANGE flg=0x4000010 (has extras) } in com.google.android.gms.wearable.node.WifiService$SavedNetworkBroadcastReceiver@b38edf9
...
08-05 09:14:51.779 17365 17365 E AndroidRuntime: Caused by: java.lang.SecurityException: WifiService: Neither user 10118 nor current process has android.permission.READ_WIFI_CREDENTIAL.
```

This does not require the watch to be currently paired to the phone - with Bluetooth off this crash still occurs.

I'm assuming that my phone is trying to automatically send the wifi password to the watch, and since Google Play Services can't access wifi credentials it crashes. I cannot turn off this syncing, it happens automatically, and thus causes GMS to crash every time I join a wifi network with credentials. It's very inconvenient, as the GMS crash also force quits some apps.

Syncing the wifi password to the watch is not strictly necessary - you can join wifi networks directly from the watch, so this feature is just for convenience. I can't speak for all users, but I'd be totally fine if my wifi passwords weren't synced between devices, I just don't want a GMS crash.

I'll note that all my other usecases with a WearOS device have worked flawlessly - thank you for all your hard work!

--- a/framework/java/android/net/wifi/WifiManager.java
+++ b/framework/java/android/net/wifi/WifiManager.java
@@ -1508,6 +1508,11 @@ public class WifiManager {
     @SystemApi
     @RequiresPermission(allOf = {ACCESS_FINE_LOCATION, ACCESS_WIFI_STATE, READ_WIFI_CREDENTIAL})
     public List<WifiConfiguration> getPrivilegedConfiguredNetworks() {
+        if (GmsCompat.isEnabled()) {
+            // READ_WIFI_CREDENTIAL is a privileged permission
+            return Collections.emptyList();
+        }
+
         try {
             ParceledListSlice<WifiConfiguration> parceledList =
                     mService.getPrivilegedConfiguredNetworks(mContext.getOpPackageName(),

